pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PublicSettings.scala

```diff
@@ -1,88 +1,56 @@
 package com.gu.pandomainauth
 
-import java.io.{ByteArrayInputStream, IOException}
-import java.util.Properties
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
 
-import akka.agent.Agent
-import com.gu.pandomainauth.PublicSettings.FunctionJob
-import okhttp3._
-import org.quartz._
-import org.quartz.impl.StdSchedulerFactory
+import com.amazonaws.services.s3.AmazonS3
+import org.slf4j.LoggerFactory
 
-import scala.collection.JavaConverters._
-import scala.collection.mutable
-import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
 
 /**
- * Class that contains the static public settings and includes mechanism for fetching the public key. The class is
- * parametrized by domain because the keys will differ for different domains. It also requires a dispatch.Http
- * instance for retrieving the data and an execution context.
- *
- * Once you have an instance you can call its start method to kick off loading the public data. In a Play app,
- * consider using the `Global` object's `onStart` method to start these calls when the application comes up. If
- * you'd rather use your own scheduler you can do so while still using the PublicSettings class by scheduling
- * your own calls to its `refresh` method can also schdule the refresh yourself using the instance's
- *
- * @param domain     The domain you would like to fetch settings for
- * @param callback   Optionally, a callback called when the data gets fetched, provided to allow you to perform logging
- * @param bucketName The name of the S3 bucket where the settings are stored
- * @param scheduler  Optionally, the quartz scheduler instance to use. It defaults to the default scheduler but
- *                   customising it may be useful if you want more control over the scheduler's lifecycle
- * @param client     Implicit instance of dispatch.Http used to make the call to fetch the public settings
- * @param ec         Implicit execution context used to fetch the settings
+ * Class that contains the static public settings and includes mechanism for fetching the public key. Once you have an
+ * instance, call the `start()` method to load the public data.
+  *
+  * @param settingsFileKey the settings file for the domain in the S3 bucket (eg local.dev.gutools.co.uk.public.settings)
+  * @param bucketName      the name of the S3 bucket (eg pan-domain-auth-settings)
+  * @param s3Client        the AWS S3 client that will be used to download the settings from the bucket
+  * @param scheduler       optional scheduler that will be used to run the code that updates the bucket
  */
-class PublicSettings(domain: String, bucketName: String, callback: Try[Map[String, String]] => Unit = _ => (),
```

Comment:
`domain` was removed since it was unused.

`callback` has also been removed since it is only used in one place across all the code in the Guardian: [s3-uploader for logging](https://github.com/guardian/s3-upload/blob/master/app/lib/PanAuth.scala#L19). The logging is useful for everyone so I've added it to the library itself.